src/osd/ECBackend.cc

```diff
@@ -1556,22 +1556,21 @@ struct ECClassicalOp : ECBackend::RMWPipeline::Op {
       DoutPrefixProvider *dpp,
       const ceph_release_t require_osd_release) final
   {
-    if (t) {
```

Comment:
Oops, I think I discovered where the t-empty ops are born:

```cpp
bool ECBackend::RMWPipeline::try_finish_rmw()
{
  // ...
    if (get_osdmap()->require_osd_release >= ceph_release_t::kraken) {
      if (op->version > get_parent()->get_log().get_can_rollback_to() &&
          waiting_reads.empty() &&
          waiting_commit.empty()) {
        // submit a dummy transaction to kick the rollforward
        auto tid = get_parent()->get_tid();
        Op *nop = &(tid_to_op_map[tid]);
        nop->hoid = op->hoid;
        nop->trim_to = op->trim_to;
        nop->roll_forward_to = op->version;
        nop->tid = tid;
        nop->reqid = op->reqid;
        waiting_reads.push_back(*nop);
      }
 ```

Reply:
Ah, I see.  Ok, add a comment to that effect so that future readers won't be confused.